exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/DrillFSDataInputStream.java

```diff
@@ -223,12 +224,28 @@ public int read(byte[] b, int off, int len) throws IOException {
     public int read(byte[] b) throws IOException {
       operatorStats.startWait();
       try {
-        return is.read(b);
+        return readBytes(b, 0, b.length);
       } finally {
         operatorStats.stopWait();
       }
     }
 
+    /**
+     * Reads up to {@code len} bytes of data from the input stream into an array of bytes.
+     * This method guarantees that regardless of the underlying stream implementation,
+     * the byte array will be populated with either {@code len} bytes or
+     * all available in stream bytes if they are less than {@code len}.
+     */
+    private int readBytes(byte[] b, int off, int len) throws IOException {
```

Comment:
I think we used a new solution to revert the `openDecompressedInputStream` in the follow case?
```
Some parsers, particularly those that read raw bytes, generate errors when passed Hadoop ZipCompressed InputStreams.
```

Reply:
`openDecompressedInputStream` doesn't behave as we want. For the case of non-compressed stream, it simply calls `open(path)`, so the issue will still be present. For the case of compressed stream, it converts all stream data to the byte array, and after that converts it to the `ByteArrayInputStream`, so we don't observe this issue, but such an approach may cause OOM, since the file may be too large.